src/usr/share/batwatch/batwatch.py

```diff
@@ -53,124 +46,150 @@ def __eq__(self, other):
+    """
+
+    def __init__(self, config):
+        self.logger = logging.getLogger(__name__)
         self.system_bus = SystemBus()
         self.upower_bus = SystemBus().get(UPOWER_BUS_NAME)
         self.config = config
 
-    def watch_the_bat(self):
-        """
-        Continuously monitor the status of all batteries attached to the device Batwatch is running
-        on, and queue a status email to a configured recipient if anything changes.
+    def start_monitoring(self):
+        """Continuously monitor the status of all batteries attached to the current system,
+        and queue a status email if any of those batteries' state changes.
         """
-        self.logger.info('Watching the battery status.')
+        self.logger.info('Monitoring the system\'s power state.')
 
         # Arbitrarily initialize the current status and detect any deviations.
-        current_status = CompositeStatus(1, FULLY_CHARGED)
+        prior_status = self._get_composite_status()
+
+        if prior_status.charge_status == DISCHARGING:
+            message = 'This system was discharging when Batwatch was initialized.'
+            self.logger.warning(message)
+            self._send_email(message)
 
         while True:
-            self.logger.trace('Running the main loop.')
-            new_status = self._get_composite_status()
-            if current_status != new_status:
-                self.logger.warn('A change in the battery status was detected. \nNEW STATUS: {} \nOLD STATUS: {}'
-                                 .format(current_status, new_status))
-                self._send_status_email(current_status, new_status)
-                current_status = new_status
+            current_status = self._get_composite_status()
+            if prior_status != current_status:
+                if prior_status < current_status:
+                    self.logger.warning('Battery state changed from %s to %s.',
+                                        prior_status, current_status)
+                else:
+                    self.logger.info('Battery state changed from %s to %s.', prior_status,
+                                     current_status)
+
+                self._send_email(self._build_email_body(prior_status, current_status))
+                prior_status = current_status
             else:
                 self.logger.trace('No changes in battery status.')
-            time.sleep(int(self.config['delay']))
 
-    def _get_composite_status(self):
-        """
-        Get status information about batteries connected to the device Batwatch is running on,
-         including the charge status, such that:
-        * If any battery is discharging, charge status is Discharging.
-          (Assume any status other than Charging or Fully Charged is equivalent to Discharging.)
-        * Charge status is Fully Charged only if all batteries are Fully Charged.
+            delay = random.uniform(0, self.config['average_delay'])
```

Comment:
Should we have an inline comment explaining why the delay is randomized?

Reply:
Why *is* the delay randomized? I only put it in because you asked me to.

Reply:
Basically to make it harder for adversaries to fingerprint our program.